src/pages/Graph/GraphRouteHandler.tsx

```diff
@@ -170,21 +175,16 @@ export class GraphRouteHandler extends React.Component<RouteComponentProps<Graph
       nodeTypeHasChanged;
 
     // update the redux store with the URL information
+    if (durationHasChanged) {
+      store.dispatch(UserSettingsActions.setDuration(nextGraphDuration));
+    }
```

Comment:
Not a blocker, but we should find a better way to do this.

The `getDerivedStateFromProps` method is turning somewhat brittle the url and redux managing. If, for any reason, React raises an update/re-render of this `GraphRouteHandler` component after dispatching a setDuration/setActiveNamespace, these lines may override the newly setted values and revert to the old ones present in the URL, before the `GraphPage` component can update the URL with the new values.

Reply:
We need to be able to change the redux state based on the URL.  In theory this should be getting called when a URL is being processed, either by a new browser entry, an F5, or a programmatic push.  In any of these cases the idea is that the URL should "rule the world" so I'm not sure if there is a problem here, as the new URL would have its values applied.  I may be missing something.  I'm open to ideas for changing things if needed.